tests/functional/export.sh

```
@@ -60,3 +62,14 @@ clearStore
 nix nario import --trusted-public-keys "$public_key" < $TEST_ROOT/exp_all
 [[ $(nix path-info --json "$outPath" | jq -r .[].signatures[]) =~ my-key: ]]
+
+# Test json listing.
+json=$(nix nario list --json < "$TEST_ROOT/exp_all")
+[[ $(printf "%s" "$json" | jq -r ".paths.\"$outPath\".deriver") = $drvPath ]]
+[[ $(printf "%s" "$json" | jq -r ".paths.\"$outPath\".contents.type") = directory ]]
+[[ $(printf "%s" "$json" | jq -r ".paths.\"$outPath\".contents.entries.foobar.type") = regular ]]
+[[ $(printf "%s" "$json" | jq ".paths.\"$outPath\".contents.entries.foobar.size") = 7 ]]
+
+json=$(nix nario list --json --no-contents < "$TEST_ROOT/exp_all")
+[[ $(printf "%s" "$json" | jq -r ".paths.\"$outPath\".deriver") = $drvPath ]]
```

Shellcheck (0.11.0)

[warning] Quote the right-hand side of = in [[ ]] to prevent glob matching.

(SC2053)